### ISSUE TITLE
Zed regression between 0.6.5.9 and 0.7.12

### ISSUE THREAD

**Issue by gw1urf** [zfsonlinux/zfs#9100]
### System information

Type | Version/Name
 --- | --- 
Distribution Name	| Debian Buster
Distribution Version	| 10
Linux Kernel	| 4.19.0-5
Architecture	| amd64
ZFS Version	| 0.7.12-2+deb10u1
SPL Version	| 0.7.12.2

### Describe the problem you're observing
In our setup, we have a disk that's marked as a spare in the pool. Because of layout requirements (redundancy across physical trays in the rack), we don't want spares to go live automatically, but would rather that they were brought live by hand after (possibly) moving to a different tray. So we're effectively using the "spare" classification for documentation.

On the previous version of zed (Debian package 0.6.5.9-5), the zed.rc file allowed control over hot spare replacement, using the ZED_SPARE_ON_CHECKSUM_ERRORS and ZED_SPARE_ON_IO_ERRORS variables. These variables are no longer present in the zed.rc 0.7.12-2 (as distributed with Debian Buster). Looking at the source code from GitHub, I can see that, rather than punting the replacement to a zedlet, it is now handled directly by zed. So far as I can see, there are no configuration options which would allow me to veto automatic replacement any more.

Obviously, there are other ways to tackle this issue, but having a "warm" spare was quite handy, and it now appears not to be an option.

### Describe how to reproduce the problem
Configure a disk as a spare in a pool. Fail out another disk. Watch Zed bring it live no matter what you do!

**Comment by behlendorf** [comment]
> but having a "warm" spare was quite handy, and it now appears not to be an option.

I agree this behavior really needs to be configurable.  We could handle this by adding a new command line option to the ZED or a configuration file for it.

Personally, I'd propose we add a new `autospare=on|off` property to the pool to control this behavior.  This would be nice and consistent with the existing `autoonline` and `autoreplace` properties, and it would allow you easily modify the behavior without the need to restart (or SIGHUP) the ZED.

In the meanwhile, if you need a fix now and are willing to rebuild the ZED you could make the following small change to disable automatic hot space replacement entirely.

```diff
diff --git a/cmd/zed/agents/zfs_retire.c b/cmd/zed/agents/zfs_retire.c
index f3dbb24..4469d9f 100644
--- a/cmd/zed/agents/zfs_retire.c
+++ b/cmd/zed/agents/zfs_retire.c
@@ -496,7 +496,7 @@ zfs_retire_recv(fmd_hdl_t *hdl, fmd_event_t *ep, nvlist_t *nvl,
                /*
                 * Attempt to substitute a hot spare.
                 */
-               (void) replace_with_spare(hdl, zhp, vdev);
+               // (void) replace_with_spare(hdl, zhp, vdev);
                zpool_close(zhp);
        }
 
```

**Comment by tsjk** [comment]
Has anything been done here? Reinstating those thresholds would be nice.

**Comment by behlendorf** [comment]
Not yet, there is some related work #13805 underway.
